tools/build_cros_gcc

```diff
@@ -1,178 +1,212 @@
 
- if test ! $2; then
-   echo "$1" | grep x86_64 1>/dev/null
-   if [ $? == 0 ]; then
-   $0 $1 x86 $3;
-   exit 0;
+ if ! [[ "$1" =~ "^[0-9]+$" ]] &>/dev/null; then
+   JOBS="$1"
+   shift
+ else
+   shift
+ fi
+ if [[ -d $1 ]] &>/dev/null; then
+   PREFIX="$1"
+   shift
+ else
+   if ! [[ "$1" =~ "^[0-9]+$" ]] &>/dev/null; then
+     JOBS="$1"
+     shift
+   else if { mkdir -p "$1" &>/dev/null; }; then
+     PREFIX="$1"
+     shift
    fi
-   echo "$1" | grep i686 1>/dev/null
-   if [ $? == 0 ]; then
-   $0 $1 x86 $3;
-   exit 0;
+ fi
+ if [[ -z "${ARCH}" ]]; then
+   if { echo "${TARGET}" | grep x86_64 1>/dev/null; }; then
+     ARCH="x86_64";
    fi
-   echo "$1" | grep armv7l 1>/dev/null
-   if [ $? == 0 ]; then
-   $0 $1 arm $3;
-   exit 0;
+   if { echo "${TARGET}" | grep i686 1>/dev/null; }; then
+     ARCH="x86";
    fi
-   echo "$1" | grep aarch64 1>/dev/null
-   if [ $? == 0 ]; then
-   $0 $1 arm64 $3;
-   exit 0;
+   if { echo "${TARGET}" | grep armv7l 1>/dev/null; }; then
+     ARCH="arm";
    fi
+   if { echo "${TARGET}" | grep aarch64 1>/dev/null; }; then
+     ARCH="arm64";
+   fi
+ fi
+ if [[ -z "${PREFIX}" ]]; then
+   PREFIX="$(crew const CREW_PREFIX | cut -d'=' -f2)"
  fi
- if test ! $3; then
-   $0 $1 $2 $(crew const CREW_PREFIX | cut -d'=' -f2)
-   exit 0
+ if [[ -z "${JOBS}" ]]; then
+   JOBS="$(nproc --all)"
  fi
 
- $0 -c
+ rm -rf \
+ ./${GLIBC} \
+ ./${GMP} \
+ ./${GCC} \
+ ./${BINUTILS} \
+ ./linux-${KERNEL_VERSION} \
+ ./${MPFR} \
+ ./${CLOOG} \
+ ./${ISL} \
+ ./${MPC} \
+ ./sha256sum
 
- echo $1 $2 $3
+ wget -c https://ftpmirror.gnu.org/gnu/binutils/${BINUTILS}.tar.xz
+ wget -c https://ftpmirror.gnu.org/gnu/gcc/${GCC}/${GCC}.tar.xz
+ wget -c https://www.kernel.org/pub/linux/kernel/v"${MAJOR_VERSION}".x/linux-${KERNEL_VERSION}.tar.xz
+ wget -c https://ftpmirror.gnu.org/gnu/libc/${GLIBC}.tar.xz
+ wget -c https://ftpmirror.gnu.org/gnu/mpfr/${MPFR}.tar.xz
+ wget -c https://ftpmirror.gnu.org/gnu/gmp/${GMP}.tar.xz
+ wget -c https://ftpmirror.gnu.org/gnu/mpc/${MPC}.tar.gz
+ wget -c http://isl.gforge.inria.fr/${ISL}.tar.bz2
+ wget -c https://www.bastoul.net/cloog/pages/download/${CLOOG}.tar.gz
 
- wget https://ftpmirror.gnu.org/gnu/binutils/$BINUTILS.tar.xz
- wget https://ftpmirror.gnu.org/gnu/gcc/$GCC/$GCC.tar.xz
- wget https://www.kernel.org/pub/linux/kernel/v"$MAJOR_VERSION".x/linux-$KERNEL_VERSION.tar.xz
- wget https://ftpmirror.gnu.org/gnu/libc/$GLIBC.tar.xz
- wget https://ftpmirror.gnu.org/gnu/mpfr/$MPFR.tar.xz
- wget https://ftpmirror.gnu.org/gnu/gmp/$GMP.tar.xz
- wget https://ftpmirror.gnu.org/gnu/mpc/$MPC.tar.gz
- wget http://isl.gforge.inria.fr/$ISL.tar.bz2
- wget https://www.bastoul.net/cloog/pages/download/$CLOOG.tar.gz
+ SUMS="5d20086ecf5752cc7d9134246e9588fa201740d540f7eb84d795b1f7a93bca86  $BINUTILS.tar.xz\n"
+ SUMS+="325adf3710ce2229b7eeb9e84d3b539556d093ae860027185e7af8a8b00a750e  $CLOOG.tar.gz\n"
+ SUMS+="832ca6ae04636adbb430e865a1451adf6979ab44ca1c8374f61fba65645ce15c  $GCC.tar.xz\n"
```

Comment:
Has this sha256 sum been updated to match the current crew gcc version?

Reply:
Yes.